python/hopsworks_common/constants.py

```diff
@@ -0,0 +1,251 @@
+#
+#   Copyright 2022 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+
+class Default:
+    """An object of this class is used in place of optional arguments in
+    cases when passing None should result in an exception or other behaviour.
+    Overwritting None with a default value is undesirable in some cases,
+    and this class solves the problem."""
+
+    def __repr__(self):
+        return "DEFAULT"
```

Comment:
Do we need to overwrite `__eq__` to check only for the instance type? Something like:
```
def __eq__(self, other):
    return isinstance(other, Default)
```
to use it like:
```
def method(self, optional=DEFAULT):
    if optional == DEFAULT:
        # use default value
```
or will DEFAULT return always the same object, and therefore match the same object ID?

Reply:
DEFAULT returns the same object, but I liked your proposal, so it was implemented.